orchagent/nhgorch.cpp

```diff
@@ -0,0 +1,1386 @@
+#include "nhgorch.h"
+#include "switchorch.h"
+#include "neighorch.h"
+#include "crmorch.h"
+#include "routeorch.h"
+#include "bulker.h"
+#include "logger.h"
+#include "swssnet.h"
+
+extern sai_object_id_t gSwitchId;
+
+extern PortsOrch *gPortsOrch;
+extern CrmOrch *gCrmOrch;
+extern NeighOrch *gNeighOrch;
+extern SwitchOrch *gSwitchOrch;
+extern RouteOrch *gRouteOrch;
+
+extern size_t gMaxBulkSize;
+
+extern sai_next_hop_group_api_t* sai_next_hop_group_api;
+extern sai_next_hop_api_t*         sai_next_hop_api;
+extern sai_switch_api_t*            sai_switch_api;
+
+unsigned int NextHopGroup::m_count = 0;
+
+/* Default maximum number of next hop groups */
+#define DEFAULT_NUMBER_OF_ECMP_GROUPS   128
+#define DEFAULT_MAX_ECMP_GROUP_SIZE     32
```

Comment:
find a common header and re-use existing macros.

Reply:
Done